scr/main/java/bittercode/util/StoreUtil.java

```diff
@@ -0,0 +1,86 @@
+package com.bittercode.util;
+
+import java.io.PrintWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import com.bittercode.model.UserRole;
+
+/*
+ * Store UTil File To Store Commonly used methods
+ */
+public class StoreUtil {
+
+    /**
+     * Check if the User is logged in with the requested role
+     */
+    public static boolean isLoggedIn(UserRole role, HttpSession session) {
+
+        return session.getAttribute(role.toString()) != null;
+    }
+
+    /**
+     * Modify the active tab in the page menu bar
+     */
+    public static void setActiveTab(PrintWriter pw, String activeTab) {
+
+        pw.println("<script>document.getElementById(activeTab).classList.remove(\"active\");activeTab=" + activeTab
+                + "</script>");
+        pw.println("<script>document.getElementById('" + activeTab + "').classList.add(\"active\");</script>");
+
+    }
+
+    /**
+     * Add/Remove/Update Item in the cart using the session
+     */
+    public static void updateCartItems(HttpServletRequest req) {
+        String selectedBookId = req.getParameter("selectedBookId");
+        HttpSession session = req.getSession();
+        if (selectedBookId != null) { // add item to the cart
+
+            // Items will contain comma separated bookIds that needs to be added in the cart
+            String items = (String) session.getAttribute("items");
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2438181340>#16fdcd</a><span> - 10/25/2024, 04:08 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Inefficient string manipulation in cart management can lead to performance issues with large numbers of items.
</div>
<div id="fix">
<b>Fix:</b> Replace string-based cart management with a more efficient data structure like HashMap or ArrayList.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -37,47 +37,41 @@ public class StoreUtil {
  public static void updateCartItems(HttpServletRequest req) {
          String selectedBookId = req.getParameter("selectedBookId");
          HttpSession session = req.getSession();
 -        if (selectedBookId != null) { // add item to the cart
 +        if (selectedBookId != null) {
 +            Map<String, Integer> cart = (Map<String, Integer>) session.getAttribute("cart");
 +            if (cart == null) {
 +                cart = new HashMap<>();
 +                session.setAttribute("cart", cart);
 +            }

 -            // Items will contain comma separated bookIds that needs to be added in the cart
 -            String items = (String) session.getAttribute("items");
              if (req.getParameter("addToCart") != null) { // add to cart
 -                if (items == null || items.length() == 0)
 -                    items = selectedBookId;
 -                else if (!items.contains(selectedBookId))
 -                    items = items + "," + selectedBookId; // if items already contains bookId, don't add it
 -
 -                // set the items in the session to be used later
 -                session.setAttribute("items", items);
 -
 -                /*
 -                 * Quantity of each item in the cart will be stored in the session as:
 -                 * Prefixed with qty_ following its bookId
 -                 * For example 2 no. of book with id 'myBook' in the cart will be
 -                 * added to the session as qty_myBook=2
 -                 */
 -                int itemQty = 0;
 -                if (session.getAttribute("qty_" + selectedBookId) != null)
 -                    itemQty = (int) session.getAttribute("qty_" + selectedBookId);
 -                itemQty += 1;
 -                session.setAttribute("qty_" + selectedBookId, itemQty);
 +                cart.put(selectedBookId, cart.getOrDefault(selectedBookId, 0) + 1);
          } else { // remove from the cart
 -                int itemQty = 0;
 -if (session.getAttribute("qty_" + selectedBookId) != null)
 -                    itemQty = (int) session.getAttribute("qty_" + selectedBookId);
 +                int itemQty = cart.getOrDefault(selectedBookId, 0);
              if (itemQty > 1) {
 -                    itemQty--;
 -                    session.setAttribute("qty_" + selectedBookId, itemQty);
 +                    cart.put(selectedBookId, itemQty - 1);
              } else {
 -                    session.removeAttribute("qty_" + selectedBookId);
 -                    items = items.replace(selectedBookId + ",", "");
 -                    items = items.replace("," + selectedBookId, "");
 -                    items = items.replace(selectedBookId, "");
 -                    session.setAttribute("items", items);
 +                    cart.remove(selectedBookId);
              }
 -                else  {
 -                    session.removeAttribute("qty_" + selectedBookId);
 -                    session.setAttribute("items", items.replace(selectedBookId, ""));
 -                }
          }
      }
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged